logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -141,6 +166,60 @@ class ConversationDataSource(
     override suspend fun deleteMember(conversationID: QualifiedIDEntity, userID: QualifiedIDEntity): Either<CoreFailure, Unit> =
         wrapStorageRequest { conversationDAO.deleteMemberByQualifiedID(conversationID, userID) }
 
+    override suspend fun createGroupConversation(name: String, members: List<Member>, options: ConverationOptions): Either<CoreFailure, Conversation> = suspending {
+        wrapStorageRequest {
+            userRepository.getSelfUser().first()
+        }.flatMap { selfUser ->
+            wrapApiRequest {
+                conversationApi.createNewConversation(
+                    conversationMapper.toApiModel(name, members, selfUser.team, options)
+                )
+            }.flatMap { conversationResponse ->
+                val teamId = selfUser.team?.let { TeamId(it) }
+                val conversationEntity = conversationMapper.fromApiModelToDaoModel(conversationResponse, groupCreation = true, teamId)
+                val conversation = conversationMapper.fromDaoModel(conversationEntity)
+
+                wrapStorageRequest {
+                    conversationDAO.insertConversation(conversationEntity)
+                }.flatMap {
+                    if (options.protocol == ConverationOptions.Protocol.PROTEUS) {
+                        persistMembersFromConversationResponse(conversationResponse)
+                    } else {
+                        persistMembersFromConversationResponseMLS(conversationResponse, members)
+                    }
+                }.flatMap {
+                    if (options.protocol == ConverationOptions.Protocol.PROTEUS) {
+                        Either.Right(conversation)
+                    } else {
+                        mlsConversationRepository.establishMLSGroup(conversation).flatMap {
+                            Either.Right(conversation)
+                        }
+                    }
```

Comment:
perhaps this establishMLSGroup thingy could be moved to a use case ?

Reply:
`establishMLSGroup()` is an internal thing we need to do before we can send messages in a MLS conversation, but the user of Kalium shouldn't have to care about that so shouldn't expose it as a use case.

Reply:
I do not  mean expose it as a use case but handel it inside the body of the usecase for example mlving it to mlsrepo and use it inside repo